tests/load/utils.py

```diff
@@ -740,41 +739,52 @@ def destinations_configs(
 @pytest.fixture(autouse=True)
 def drop_pipeline(request, preserve_environ) -> Iterator[None]:
     # NOTE: keep `preserve_environ` to make sure fixtures are executed in order``
+    # enable activation history (for the main thread) to be able to drop pipelines active during test
+    Container()[PipelineContext].enable_activation_history = True
     yield
-    if "no_load" in request.keywords:
-        return
     try:
-        drop_active_pipeline_data()
+        if "no_load" in request.keywords:
+            # always deactivate
+            Container()[PipelineContext].deactivate()
+            Container()[PipelineContext].clear_activation_history()
```

Comment:
hm, maybe we want to keep the activation history here so we can clean up with `drop_active_pipeline_data` after a bunch of "no_load" tests have run without having to store the pipelines?

Reply:
I added this because most of `no_load` tests produce empty or fake pipelines ie. with destinations that are not instantiated. so they were leaking into subsequent tests and breaking them on cleanup. I'll keep it util we have a good case